### PR TITLE
chore(linter): fix some of the warnings from gas linter

### PR DESCRIPTION
**Pull request by mangalaman93**
this is the linter https://github.com/securego/gosec

### CODE REVIEW

**Comment by anurags92** [comment]
@mangalaman93 Can you add what's gas linter (probably a link as well)?

**Comment by skrdgraph** [comment]
> @mangalaman93 Can you add what's gas linter (probably a link as well)?

`gas` is a security linter. 

@anurags92 this explains all linters https://golangci-lint.run/usage/linters/ (I recommend spending about 15mins doing a high level reading) & compare with our `golangci yaml` 

Also this was something I had bookmarked https://github.com/golangci/awesome-go-linters (may have deprecated linters).

**Review by skrdgraph — APPROVED** [review]
The coverage upload failed, there was probably a glitch in network or an issue on coveralls. The test will need a re-trigger.

I also think this will be interesting to observe our logs on v23.

**Comment by mangalaman93** [comment]
Thanks Sudhish, yeah, will need to keep an eye out. I didn't see anything interesting during my testing locally on the laptop.

File: graphql/e2e/common/subscription.go [hunk 1]
```diff
@@ -85,7 +86,6 @@ func NewGraphQLSubscription(url string, req *schema.Request, subscriptionPayload
 
 	msg := operationMessage{}
 	if err = conn.ReadJSON(&msg); err != nil {
-		conn.Close()
```
Comment (anurags92) [inline graphql/e2e/common/subscription.go:88]: Are we sure, we are not leaking the connection here? It is not obvious from the code.

**Review by anurags92 — DISMISSED** [review]

File: graphql/e2e/common/subscription.go (hunk 1 above)
Comment (mangalaman93) [reply graphql/e2e/common/subscription.go:88]: We probably are but this is a test and we are not doing any worse than what we were doing before this PR.

**Review by anurags92 — APPROVED** [review]

**Review by harshil-goel — APPROVED** [review]

**Pull request merged**